platform/mbed_retarget.cpp

```diff
@@ -880,6 +880,9 @@ extern "C" WEAK void __iar_file_Mtxinit(__iar_Rmtx *mutex) {}
 extern "C" WEAK void __iar_file_Mtxdst(__iar_Rmtx *mutex) {}
 extern "C" WEAK void __iar_file_Mtxlock(__iar_Rmtx *mutex) {}
 extern "C" WEAK void __iar_file_Mtxunlock(__iar_Rmtx *mutex) {}
+#if (__IAR_SYSTEMS_ICC__ >= 8)
+extern "C" WEAK void *__aeabi_read_tp (void) {}
```

Comment:
why is this suddenly required, and what about the other requirements? As for any thread, there is more to implement than just read_tp ? See http://netstorage.iar.com/SuppDB/Public/UPDINFO/012120/arm/doc/infocenter/iccarm.ENU.html